grpc/src/main/java/com/replica/RouterImpl.java

```diff
@@ -295,38 +296,65 @@ public void routePt(PtRouteRequest request, StreamObserver<PtRouteReply> respons
                     continue;
                 }
 
-                // Add stable edge IDs to PT legs
-                List<Trip.Leg> ptLegs = path.getLegs().stream()
-                        .filter(leg -> leg.type.equals("pt"))
-                        .map(leg -> getCustomPtLeg((Trip.PtLeg)leg))
-                        .collect(toList());
-
-                // Add stable edge IDs to walk legs
-                List<Trip.Leg> walkLegs = path.getLegs().stream()
-                        .filter(leg -> leg.type.equals("walk"))
-                        .collect(toList());
-
-                Trip.WalkLeg firstLeg = (Trip.WalkLeg) walkLegs.get(0);
-                Trip.WalkLeg lastLeg = (Trip.WalkLeg) walkLegs.get(1);
-
-                List<String> lastLegStableIds = lastLeg.details.get("stable_edge_ids").stream()
-                        .map(idPathDetail -> (String) idPathDetail.getValue())
-                        .filter(id -> id.length() == 20)
-                        .collect(toList());
-
-                // The first leg contains stable IDs for both walking legs for some reason,
```

Comment:
One note here - this hacky code was removed in this PR, and I was a bit worried about it, but I checked and verified that it no longer seems to be needed. It seems to be the case that the bug we were addressing here no longer is a problem in graphhopper (I added a unit test just to be sure that all stable edge IDs across all legs are unique as well).

@michaz , does this sound right to you?